develop/entities/first-entity.md

```diff
@@ -0,0 +1,163 @@
+
+This tutorial will walk you through the creation of a `PathfinderEntity`, used by most mobs with pathfinding like `Zombie` and `Villager` alike.
+
+## Preparing Your First Entity {#preparing-your-first-entity}
+
+The first step in creating a custom entity is defining its class and registering it with the game.
+
+We'll create the class `MiniGolemEntity` for our entity, and start by giving it attributes. [Attributes](attributes) decide the Maximum Health, Movement Speed & Tempt Range of the entity.
+
+@[code transcludeWith=:::registerclass](@/reference/latest/src/main/java/com/example/docs/entity/MiniGolemEntity.java)
+
+To register your entity, it's recommended to create a separate class, `ModEntityTypes`, where you register any and all entity types, set their dimensions, and register their attributes.
+
+@[code transcludeWith=:::types](@/reference/latest/src/main/java/com/example/docs/entity/ModEntityTypes.java)
+
+## Adding Goals {#adding-goals}
+
+Goals are the system that handle an entity's objective/aim, providing them with a defined set of behaviour. Goals have a certain priority, ranging from `0` being the highest and subsequently reducing in priority as the number increases.
+
+To add goals to the entity, you need to create a `registerGoals` method in your entity's class that defines the goals for the entity.
+
+@[code transcludeWith=:::goals](@/reference/latest/src/main/java/com/example/docs/entity/MiniGolemEntity.java)
+
+::: info
+
+1. `TemptGoal` - The entity is attracted towards a player holding an item.
+2. `RandomStrollGoal` - Walks/Wanders around the world.
+3. `LookAtPlayerGoal` - To look at the `Cow` entity.
+4. `RandomLookAroundGoal` - To look in random directions.
+
+:::
+
+## Creating Rendering {#creating-rendering}
+
+Rendering refers to the process of converting game data such as blocks, entities, and environments into visual representations displayed on the player's screen. This involves determining how objects are illuminated, shaded, and textured.
+
+::: tip
+Entity rendering is always handled on the client side. The server manages the entity's logic and behavior, while the client is responsible for displaying the entity's model, texture, and animations.
+:::
+
+Rendering has multiple steps involving their own classes, but we'll start with the `EntityRenderState` class.
+
+@[code transcludeWith=:::entitystate](@/reference/latest/src/client/java/com/example/docs/entity/state/MiniGolemEntityRenderState.java)
+
+The render state determines how the entity is visually represented, including animation states such as movement and idle behaviors.
+
+### Setting up the Model {#setting-up-model}
+
+The `MiniGolemEntityModel` class defines how your entity looks by describing its shape and parts. Models are generally created in third-party tools like [Blockbench](https://web.blockbench.net), rather than being written by hand.
```

Comment:
```suggestion
The `MiniGolemEntityModel` class defines how your entity looks by describing its shape and parts. Models are generally created in third-party tools like [Blockbench](https://web.blockbench.net), rather than being written by hand. Nonetheless, this tutorial will go through a manual example to show you how it works.
```